### PR TITLE
fixed bug where SNP and indel pseudocounts were swapped

### LINKED ISSUE #5885: AFCalculationResult uses snpPseudocount for insertions and indelPseudocount for SNPs and deletions
### Affected tool(s) or class(es)
org.broadinstitute.hellbender.tools.walkers.genotyper.afcalc.AlleleFrequencyCalculator

### Affected version(s)
Latest master branch as of April 16, 2019

### Description 
At, https://github.com/broadinstitute/gatk/blob/master/src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/afcalc/AlleleFrequencyCalculator.java#L66

For insertions, `priorPseudocount` is set to `snpPseudocount`. With SNPs and deletions, `priorPseudocount` is set to `indelPseudocount`.

SNPs should have the `priorPseudocount` set to `snpPseudocount`. Indels should have `priorPseudocount` set to `indelPseudocount`.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #5885.  @ldgauthier Fortunately the change in output is tiny, as you can see in the changed integration test VCFs.

### CODE REVIEW

File: src/test/resources/org/broadinstitute/hellbender/tools/walkers/CombineGVCFs/newMQcalc.combined.g.vcf [hunk 1]
```diff
@@ -11542,7 +11542,7 @@
 20	10099258	.	G	<NON_REF>	.	.	END=10099260	GT:DP:GQ:MIN_DP:PL	./.:30:81:28:0,81,1132	./.:32:87:32:0,87,1305
 20	10099261	.	A	<NON_REF>	.	.	END=10099264	GT:DP:GQ:MIN_DP:PL	./.:30:81:28:0,81,1132	./.:34:90:33:0,90,1350
 20	10099265	.	C	<NON_REF>	.	.	END=10099269	GT:DP:GQ:MIN_DP:PL	./.:30:81:28:0,81,1132	./.:35:99:34:0,99,1485
-20	10099270	.	A	<NON_REF>	.	.	.	GT:DP:GQ:MIN_DP:PL	./.:30:37:30:0,37,1019	./.:34:90:32:0,90,1033
+20	10099270	.	A	C,<NON_REF>	.	.	BaseQRankSum=-2.866e+00;DP=68;ExcessHet=3.01;MQRankSum=-2.691e+00;RAW_MQandDP=114878,36;ReadPosRankSum=-1.380e-01	GT:AD:DP:GQ:MIN_DP:PL:SB	./.:27,3,0:30:60:.:0,60,1084,81,1092,1113:19,8,0,3	./.:.:34:90:32:0,90,1033,90,1033,1033
```
[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/test/resources/org/broadinstitute/hellbender/tools/walkers/CombineGVCFs/newMQcalc.combined.g.vcf]: This scared me, too.  Fortunately, while this is an output for `CombineGVCFSIntegrationTest`, it is only an input for `GenotypeGVCFsIntegrationTest`.  That's why there's no QUAL.  The site is correctly found to be monomorphic.  Nonetheless, is it worth looking into the change in `CombineGVCFs`?

File: src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf [hunk 2]
```diff
@@ -1316,73 +1316,73 @@
...
+20	10099250	.	G	A,<NON_REF>	720.64	.	AS_RAW_BaseQRankSum=|3.1,1|NaN;AS_RAW_MQ=57600.00|61406.00|0.00;AS_RAW_MQRankSum=|-2.3,1|NaN;AS_RAW_ReadPosRankSum=|0.5,1|NaN;AS_SB_TABLE=12,4|11,10|0,0;BaseQRankSum=3.170;DP=39;ExcessHet=3.0103;MLEAC=1,0;MLEAF=0.500,0.00;MQRankSum=-2.272;RAW_MQandDP=126206,39;ReadPosRankSum=0.583	GT:AD:DP:GQ:PL:SB	0/1:16,21,0:37:99:728,0,424,776,487,1263:12,4,11,10
 20	10099251	.	A	<NON_REF>	.	.	END=10099269	GT:DP:GQ:MIN_DP:PL	0/0:30:81:28:0,81,1132
-20	10099270	.	A	<NON_REF>	.	.	END=10099270	GT:DP:GQ:MIN_DP:PL	0/0:30:37:30:0,37,1019
+20	10099270	.	A	C,<NON_REF>	0	.	AS_RAW_BaseQRankSum=|-2.9,1|NaN;AS_RAW_MQ=92210.00|6058.00|0.00;AS_RAW_MQRankSum=|-2.7,1|NaN;AS_RAW_ReadPosRankSum=|-0.2,1|NaN;AS_SB_TABLE=19,8|0,3|0,0;BaseQRankSum=-2.866;DP=36;ExcessHet=3.0103;MLEAC=0,0;MLEAF=0.00,0.00;MQRankSum=-2.691;RAW_MQandDP=114878,36;ReadPosRankSum=-0.138	GT:AD:DP:GQ:PL:SB	0/0:27,3,0:30:60:0,60,1084,81,1092,1113:19,8,0,3
```
Comment (ldgauthier) [inline src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf]: The root change is here.  I think it's worth running through the debugger real quick.

Comment (davidbenjamin) [reply src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf]: Thanks for pointing me there.  I will look into it.

Comment (davidbenjamin) [reply src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf]: Okay, it's nothing special, just a value that eked over a threshold with the new prior.
Our `configuration.genotypeArgs.STANDARD_CONFIDENCE_FOR_CALLING` is `-0.0`, but we use a miniscule positive value in practice:
```
public boolean passesThreshold(final Allele allele, final double phredScaleQualThreshold) {
        Utils.nonNull(allele);
        return getLog10PosteriorOfAlleleAbsent(allele) + EPSILON < QualityUtils.qualToErrorProbLog10(phredScaleQualThreshold);
    }
```
where `EPSILON` is 10^-10.  The C alt allele at this site has, with the correct SNP prior, a probability of 7*10^-10, so it just barely makes it.

So, there's nothing amiss here.  A case could be made for increasing our `EPSILON`, but that's independent of this change.

Comment (ldgauthier) [reply src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf]: So many epsilons, so little time...  I'm satisfied with this explanation.  :shipit:

**Pull request merged**